sink/datadog.go

```diff
@@ -130,7 +160,11 @@ func (s *Datadog) Send(ctx context.Context, m *blip.Metrics) error {
 	if n == 0 {
 		return fmt.Errorf("no Blip metrics were collected")
 	}
-	dp := make([]datadogV2.MetricSeries, n)
+	var dp []datadogV2.MetricSeries
+
+	if s.api() {
```

Comment:
Don't use superfluous func like this (specifically here and generally speaking). It's just an unnecessary redirection for future eng. Instead,
```go
// Prealloc DD DP for API 
if !s.dogstatsd {
}
```
Plus, you do this below :-) `if s.dogStatsD {`

Reply:
This is probably a bit too extreme, but superfluous well-named small functions often helps with reducing cognitive load of the future readers, as they won't have to do the expression evaluation in their head. For high level public functions, the general flow is often what you are interested in, not the details. I will inline it.